File: programmers/pg_lv1_추억점수.js
function solution(name, yearning, photo) {
  let matchMap = new Map();
  let result = [];

  for (i = 0; i < name.length; i++) {
    matchMap.set(name[i], yearning[i]);
  }

  photo.forEach((arr) => {
    let sum = 0;
    arr.forEach((ele) => {
      if (matchMap.has(ele)) sum += matchMap.get(ele);
    });
    result.push(sum);
  });
  return result;
}

File: python/programmers/pg_lv1_추억점수.py
def solution(name, yearning, photo):
    result = []
    memories = dict()
    
    for i in range(len(name)):
        memories[name[i]] = yearning[i]
    
    for j in photo:
        sum = 0
        for k in j:
            if k in memories: sum += memories.get(k)
        result.append(sum)
        
    return result